readme.md

```diff
@@ -1,848 +1,1214 @@
+			string: 'a string',
+			// An array property.
+			array: [ 1, 2, 3, 4 ],
+			// An object property.
+			object: {
+				lang: 'en-Us'
+			},
+
+			getData: function() {
+				// Get the current value of `data`.
+				return data;
+			},
+
+			setData: function( value ) {
+				this.value = value;
+			}
+		};
+	} )();
 
-    ```
+	// Other things might happen here.
+
+	// Expose our module to the global object.
+	CKEDITOR.module = Module;
+} )();
+```
+#### 2.5.2. Practical constructor
 
+```javascript
+( function() {
+	var fooBar = CKEDITOR.tools.createClass( {
+		$: function( foo ) {
+			this.foo = foo;
+		},
+		
+		proto: {
+			getFoo: function() {
+				return this.foo;
+			},
+
+			setFoo: function( val ) {
+				this.foo = val;
+			}
+		}
+	} );
+
+	// Expose our constructor to the global object.
+	CKEDITOR.fooBar = fooBar;
+} )();
+```
+
+#### 2.5.3. Prototype extension
+
+```javascript
+( function() {
+	var fooBar = CKEDITOR.tools.extend( FooBar.prototype, {
+		getFoo: function() {
+			return this.foo;
+		};
+
+		setFoo: function( val ) {
+			this.foo = val;
+		};
+	} );
+
+	// Expose our constructor to the global object.
+	CKEDITOR.fooBar = fooBar;
+} )();
+```
+
+### 2.6. Naming
+
+You are not a human code compiler/compressor, so don't try to be one.
+
+The following code is an example of egregious naming:
+
+```javascript
+function q(s) {
+	return document.querySelectorAll(s);
+}
+var i,a=[],els=q("#foo");
+for(i=0;i<els.length;i++){a.push(els[i]);}
+```
+
+Here's the same piece of logic, but with kinder, more thoughtful naming (and a readable structure):
+```javascript
+
+function query( selector ) {
+  return document.querySelectorAll( selector );
+}
+
+var elements = [],
+	matches = query( '#foo' ),
+	length = matches.length;
+
+for ( var i = 0; i < length; i++ ) {
+	elements.push( matches[ i ] );
+}
+```
+
+#### 2.6.1. Methods and functions
+
+Methods and functions are **almost always** verbs/actions in `camelCase` notation:
+
+```javascript
+// Good
+execute();
+this.getNextNumber();
+
+// Bad
+this.editable();
+this.status();
+```
+
+The exception for the above are get/set methods that are used very often, where the first parameter is the thing to get/set and the second is the optional value (set with it and get without):
+
+```javascript
+this.data( 'name' );  // get
+this.data( 'name', 'John' );  // set
+
+this.attr( 'title' );  // get
+this.attr( 'title', 'Flying plane' );  // set
+```
+
+#### 2.6.2. Properties and variables
+
+Properties and variables are **almost always** nouns in `camelCase` notation:
+
+```javascript
+var editor;
+this.name;
+this.editable;
+```
+Boolean properties and variables are **always** prefixed by an auxiliary verb:
+
+```javascript
+this.isDirty;
+this.hasChildren;
+this.isFake;
+```
+Properties that require computation can use getters/setters and are named as nouns instead of having them as getXXX() and setXXX() methods when they are programmatically and conceptually simple:
```

Comment:
```suggestion
Properties that require computation can use getters/setters and are named as nouns instead of having them as `getXXX()` and `setXXX()` methods when they are programmatically and conceptually simple:
```